src/Json.Schema/SchemaTypeConverter.cs

```diff
@@ -34,8 +34,7 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                 }
                 else
                 {
-                    serializer.CaptureError(jToken, ErrorNumber.InvalidTypeString, typeString);
```

Comment:
>serializer.CaptureError(jToken, ErrorNumber.InvalidTypeString, typeString); [](start = 20, length = 75)

My suggestion is that you replace this call with a call to a helper that aggregates all the collected errors. Then in a post-processing step, detect that you have a bundle of errors accumulated, format them into a single exception and throw it.

Reply:
That sounds reasonable. The helper will have to be a global singleton because these throws are spread across classes.

---
In reply to: [307554712](https://github.com/microsoft/jschema/pull/123#discussion_r307554712) [](ancestors = 307554712)

Reply:
And that presents a problem because the unit tests run in parallel and I get intermittent test failures do to the singleton error aggregator being in an invalid state. Still working on it...

---
In reply to: [307555034](https://github.com/microsoft/jschema/pull/123#discussion_r307555034) [](ancestors = 307555034,307554712)

Reply:
i am interested in your test parallelization problem when you debug it. We have many patterns in the SARIF SDK where we set static variables in product code and clear same in a finalizer. I'm wondering whether these tests are potentially vulnerable to failures such as you're experiencing (i.e., things may be working now simply because there's no contention around these statics)

---
In reply to: [307807552](https://github.com/microsoft/jschema/pull/123#discussion_r307807552) [](ancestors = 307807552,307555034,307554712)

Reply:
That is possible. See my latest iteration, which addresses the error accumulation problem _almost_ completely (but not in this class, or in the `MustBeStringConverter` class, and I will explain why when you're ready) and also addresses the parallelization problem by removing the Singleton behavior from several classes.

---
In reply to: [307820380](https://github.com/microsoft/jschema/pull/123#discussion_r307820380) [](ancestors = 307820380,307807552,307555034,307554712)

Reply:
Actually the explanation is simple, so I'll just say it: The `SchemaTypeConverter` and the `MustBeStringConverter` are instantiated by way of `JsonConverter` attributes on properties of the `JsonSchema` class -- so I don't have a chance to inject an error accumulator object into their ctors, as I do for all the other converters.

---
In reply to: [307827622](https://github.com/microsoft/jschema/pull/123#discussion_r307827622) [](ancestors = 307827622,307820380,307807552,307555034,307554712)

Reply:
This is ready for your further review. I haven't found a good way to bring `SchemaTypeConverter` and `MustBeString` converter into the "error accumulation" fold.

---
In reply to: [307829197](https://github.com/microsoft/jschema/pull/123#discussion_r307829197) [](ancestors = 307829197,307827622,307820380,307807552,307555034,307554712)

Reply:
Can you use a thread static singleton pattern? This should solve your test parallelization issue yes?

---
In reply to: [307829197](https://github.com/microsoft/jschema/pull/123#discussion_r307829197) [](ancestors = 307829197,307827622,307820380,307807552,307555034,307554712)

Reply:
Yes! This is now ready for review.

---
In reply to: [307889723](https://github.com/microsoft/jschema/pull/123#discussion_r307889723) [](ancestors = 307889723,307829197,307827622,307820380,307807552,307555034,307554712)

Reply:
Wahoo!

---
In reply to: [307909800](https://github.com/microsoft/jschema/pull/123#discussion_r307909800) [](ancestors = 307909800,307889723,307829197,307827622,307820380,307807552,307555034,307554712)